packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-editor.vue

```diff
@@ -0,0 +1,217 @@
+						<InputNumber
+							class="p-inputtext-sm"
+							inputId="numericInput"
+							mode="decimal"
+							v-model="paramValue"
+							@update:model-value="updateParameter(paramValue)"
+						/>
+					</div>
+				</div>
+				<div v-else-if="fieldType === 'initial'" class="form-section">
+					<h4>{{ currentInitial?.target }}</h4>
+					<div>
+						<h3>Expression</h3>
+						<InputText
+							class="p-inputtext-sm"
+							v-model="initialExpression"
+							@update:model-value="updateInitial(initialExpression)"
+						/>
+					</div>
+				</div>
+				<h4 v-else>Select a field to edit</h4>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup lang="ts">
+import _ from 'lodash';
+import { computed, ref, watch } from 'vue';
+import Accordion from 'primevue/accordion';
+import AccordionTab from 'primevue/accordiontab';
+import Button from 'primevue/button';
+import InputNumber from 'primevue/inputnumber';
+import InputText from 'primevue/inputtext';
+import { Model, Initial, ModelParameter } from '@/types/Types';
+import {
+	getUnstratifiedParameters,
+	getUnstratifiedInitials
+} from '@/model-representation/petrinet/mira-petri';
+import { getStratificationType } from '@/model-representation/petrinet/petrinet-service';
+
+const props = defineProps<{
+	model: Model;
+	initials?: Initial[];
+	parameters?: ModelParameter[];
+}>();
+
+const emit = defineEmits(['update-param', 'update-initial']);
+
+const stratifiedModelType = computed(() => getStratificationType(props.model));
+
+const parameters = computed<Map<string, string[]>>(() => getUnstratifiedParameters(props.model));
+const currentParam = ref<ModelParameter>();
+const paramValue = ref<number>();
+
+const initials = computed<Map<string, string[]>>(() => getUnstratifiedInitials(props.model));
+const currentInitial = ref<Initial>();
+const initialExpression = ref<string>();
+
+const fieldType = ref<string>();
+const selectedField = ref<string>();
+
+const getClickedField = (type: string, field: string) => {
+	fieldType.value = type;
+	selectedField.value = field;
+
+	if (type === 'parameter') {
+		currentParam.value = props.parameters?.find((p) => p.id === field);
+		paramValue.value = currentParam.value?.value;
+	} else if (type === 'initial') {
+		currentInitial.value = props.initials?.find((i) => i.target === field);
+		initialExpression.value = currentInitial.value?.expression;
+	}
+};
+
+const updateParameter = (paramVal?: number) => {
+	if (paramVal) {
+		emit('update-param', { ...currentParam.value, value: paramVal });
+	}
+};
+
+const updateInitial = (initialExp?: string) => {
+	if (initialExp) {
+		emit('update-initial', { ...currentInitial.value, expression: initialExp });
+	}
+};
+
+// When output changes via the dropdown, force the editor to update
+watch(
+	[() => props.initials, () => props.parameters],
+	() => {
+		if (fieldType.value && selectedField.value) {
+			getClickedField(fieldType.value, selectedField.value);
+		}
+	},
+	{ immediate: true }
+);
+</script>
```

Comment:
Can we do a bit or reorganization for the `<script>` section to make things more idiomatic? 

We typically have the pattern of:
- imports
- props/emits declarations
- refs/computed declarations
- functions
- watchers
- lifecycle-hooks

You have things a bit more intermingled.

Reply:
Sorry, posted on wrong file